go-controller/pkg/libovsdbops/template_var.go

```diff
@@ -0,0 +1,110 @@
+// 'template' variable and returns the corresponding ops.
+func CreateOrUpdateChassisTemplateVarOps(nbClient libovsdbclient.Client,
+	ops []libovsdb.Operation, template *nbdb.ChassisTemplateVar) ([]libovsdb.Operation, error) {
+
+	modelClient := newModelClient(nbClient)
+	return modelClient.CreateOrUpdateOps(ops, operationModel{
+		Model: template,
+		ModelPredicate: func(item *nbdb.ChassisTemplateVar) bool {
+			return item.Chassis == template.Chassis
+		},
+		OnModelMutations: []interface{}{&template.Variables},
+		ErrNotFound:      false,
+		BulkOp:           false,
+	})
+}
+
+// deleteChassisTemplateVarVariablesOps removes the variables listed as
+// keys of 'template.Variables' and returns the corresponding ops.
+// It applies the mutation to all records that are selected by 'predicate'.
+func deleteChassisTemplateVarVariablesOps(nbClient libovsdbclient.Client,
+	ops []libovsdb.Operation, template *nbdb.ChassisTemplateVar,
+	predicate chassisTemplateVarPredicate) ([]libovsdb.Operation, error) {
+
+	deleteTemplate := &nbdb.ChassisTemplateVar{
+		Chassis:   template.Chassis,
+		Variables: map[string]string{},
+	}
+	for name := range template.Variables {
+		deleteTemplate.Variables[name] = ""
+	}
+	modelClient := newModelClient(nbClient)
+	return modelClient.DeleteOps(ops, operationModel{
+		Model:            deleteTemplate,
+		ModelPredicate:   predicate,
+		OnModelMutations: []interface{}{&deleteTemplate.Variables},
+		ErrNotFound:      false,
+		BulkOp:           false,
+	})
+}
+
+// DeleteChassisTemplateVarVariablesOps removes all variables listed as
+// keys of 'template.Variables' from the record matching the same chassis
+// as 'template'. It returns the corresponding ops.
+func DeleteChassisTemplateVarVariablesOps(nbClient libovsdbclient.Client,
+	ops []libovsdb.Operation, template *nbdb.ChassisTemplateVar) ([]libovsdb.Operation, error) {
+
+	return deleteChassisTemplateVarVariablesOps(nbClient, ops, template,
+		func(item *nbdb.ChassisTemplateVar) bool {
+			return item.Chassis == template.Chassis
+		})
+}
+
+// DeleteChassisTemplateVarVariables removes the variables listed as
+// keys of 'template.Variables' and commits the transaction to the
+// database. It applies the mutation to all records that are selected by
+// 'predicate'.
+func DeleteChassisTemplateVarVariables(nbClient libovsdbclient.Client, template *nbdb.ChassisTemplateVar, predicate chassisTemplateVarPredicate) error {
+	ops, err := deleteChassisTemplateVarVariablesOps(nbClient, nil, template, predicate)
+	if err != nil {
+		return err
+	}
+
+	_, err = TransactAndCheck(nbClient, ops)
+	return err
+}
+
+// DeleteChassisTemplateVar deletes all complete Chassis_Template_Var
+// records matching 'templates'.
+func DeleteChassisTemplateVar(nbClient libovsdbclient.Client, templates ...*nbdb.ChassisTemplateVar) error {
+	opModels := make([]operationModel, 0, len(templates))
+	for _, template := range templates {
```

Comment:
This is a peculiarity of go: `template` is a loop variable and it's memory is re-used throughout the loop as an optimization. That means that when combined with the predicates below when they run (imagine that what golang does is taking the address of `template` for the predicate), will always use the last template.

Compare this https://go.dev/play/p/yB-knGBQ4mN to this https://go.dev/play/p/FVJi6kxW2hU

So you have two options
```
for _, template := range templates {
    template := template
```
or
```
for i := range templates {
    template := templates[i]
```

Reply:
Oh, my bad.  I'll go for the latter.